community/sample-apps/nextjs-postgres-graphql/config.js

```diff
@@ -1,10 +1,30 @@
-import { withData } from 'next-apollo'
-import { HttpLink } from 'apollo-link-http'
+import withApollo from "next-with-apollo";
+import { ApolloClient, ApolloProvider, InMemoryCache } from "@apollo/client";
 
-const config = {
-  link: new HttpLink({
-    uri: 'http://localhost:8080/v1/graphql', // <- Configure GraphQL Server URL (must be absolute)
-  })
-}
+// Learn more about `@apollo/client` here - https://www.apollographql.com/docs/react/why-apollo
+// Learn more about `next-with-apollo` here - https://github.com/lfades/next-with-apollo
 
-export default withData(config)
+// creating the Apollo Client
+const client = new ApolloClient({
+  headers: {
+    "x-hasura-admin-secret": process.env.NEXT_PUBLIC_HASURA_ADMIN_SECRET
```

Comment:
Admin secret should not be used on the frontend client. Can you remove this and replace it with either Authorization header or do not have authentication for this example?

Reply:
Hey @praveenweb! Would you guide me to do this?

Reply:
@amit-ksh - Sure. Instead of sending the x-hasura-admin-secret header, remove that and configure the unauthenticated access at the Hasura side. Here's the docs on how to configure that with an additional role to be setup.

Reply:
https://hasura.io/docs/latest/auth/authentication/unauthenticated-access/

Reply:
Done 👍